presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -140,6 +141,13 @@ public void testGenericLiteral()
         assertGenericLiteral("foo");
     }
 
+    @Test
+    public void testBinaryLiteral()
+        throws Exception
+    {
+        assertExpression("x 'a  b'", new BinaryLiteral("ab"));
```

Comment:
test empty binary literal
test invalid cases (even length, illegal character, space between x and quote)